### PR TITLE
Added migrations to add raDate to projects, and raActive and raCompul…

### PR DESCRIPTION
**Pull request by joefitter**
…sory to projects

### CODE REVIEW

File: migrations/20200708164116_add_ra_date_to_projects.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+exports.up = function(knex) {
+  return knex.schema.table('projects', table => {
+    table.dateTime('ra_date').nullable();
```
Comment (lennym) [inline migrations/20200708164116_add_ra_date_to_projects.js:4]:
```suggestion
    table.date('ra_date').nullable();
```

We don't need time precision for this.

File: schema/project.js [hunk 2]
```diff
@@ -45,6 +45,7 @@ class Project extends BaseModel {
         issueDate: { type: ['string', 'null'], format: 'date-time' },
         expiryDate: { type: ['string', 'null'], format: 'date-time' },
         revocationDate: { type: ['string', 'null'], format: 'date-time' },
+        raDate: { type: ['string', 'null'], pattern: date.yearMonthDay },
```
Comment (lennym) [inline schema/project.js:48]: Oh. I've fucked it here haven't I?

There's no good reason for this to be inconsistent.

Comment (joefitter) [reply schema/project.js:48]: lol

Comment (joefitter) [reply schema/project.js:48]: we use YYYY-MM-DD for other dates like DOB etc

**Pull request merged**